x-pack/plugins/infra/public/components/alerting/logs/expression_editor/criterion_preview_chart.tsx

```diff
@@ -0,0 +1,330 @@
+  chartAlertParams: GetLogAlertsChartPreviewDataAlertParamsSubset;
+}
+
+const CriterionPreviewChart: React.FC<ChartProps> = ({
+  buckets,
+  context,
+  sourceId,
+  threshold,
+  chartAlertParams,
+}) => {
+  const isDarkMode = context.uiSettings?.get('theme:darkMode') || false;
+
+  const {
+    getChartPreviewData,
+    isLoading,
+    hasError,
+    chartPreviewData: series,
+  } = useChartPreviewData({
+    context,
+    sourceId,
+    alertParams: chartAlertParams,
+    buckets,
+  });
+
+  useMount(() => {
+    getChartPreviewData();
+  });
+
+  useDebounce(
+    () => {
+      getChartPreviewData();
+    },
+    500,
+    [getChartPreviewData]
+  );
+
+  const isStacked = false;
+
+  const { timeSize, timeUnit, groupBy } = chartAlertParams;
+
+  const isGrouped = groupBy && groupBy.length > 0 ? true : false;
+
+  const isAbove =
+    threshold && threshold.comparator
+      ? [Comparator.GT, Comparator.GT_OR_EQ].includes(threshold.comparator)
+      : false;
+
+  const isBelow =
+    threshold && threshold.comparator
+      ? [Comparator.LT, Comparator.LT_OR_EQ].includes(threshold?.comparator)
+      : false;
+
+  // For grouped scenarios we want to limit the groups displayed, for "isAbove" thresholds we'll show
+  // groups with the highest doc counts. And for "isBelow" thresholds we'll show groups with the lowest doc counts.
+  const filteredSeries = useMemo(() => {
+    if (!isGrouped) {
+      return series;
+    }
+
+    const sortedByMax = series.sort((a, b) => {
+      const aMax = Math.max(...a.points.map((point) => point.value));
+      const bMax = Math.max(...b.points.map((point) => point.value));
+      return bMax - aMax;
+    });
+    const sortedSeries = (!isAbove && !isBelow) || isAbove ? sortedByMax : sortedByMax.reverse();
+    return sortedSeries.slice(0, GROUP_LIMIT);
+  }, [series, isGrouped, isAbove, isBelow]);
+
+  const barSeries = useMemo(() => {
+    return filteredSeries.reduce<Array<{ timestamp: number; value: number; groupBy: string }>>(
+      (acc, serie) => {
+        const barPoints = serie.points.reduce<
+          Array<{ timestamp: number; value: number; groupBy: string }>
+        >((pointAcc, point) => {
+          return [...pointAcc, { ...point, groupBy: serie.id }];
+        }, []);
+        return [...acc, ...barPoints];
+      },
+      []
+    );
+  }, [filteredSeries]);
+
+  const lookback = timeSize * buckets;
+  const hasData = series.length > 0;
+  const { yMin, yMax, xMin, xMax } = getDomain(filteredSeries, isStacked);
+  const chartDomain = {
+    max: threshold && threshold.value ? Math.max(yMax, threshold.value) * 1.1 : yMax * 1.1, // Add 10% headroom.
+    min: threshold && threshold.value ? Math.min(yMin, threshold.value) : yMin,
+  };
+
+  if (threshold && threshold.value && chartDomain.min === threshold.value) {
+    chartDomain.min = chartDomain.min * 0.9; // Allow some padding so the threshold annotation has better visibility
+  }
+
+  const opacity = 0.3;
```

Comment:
I would rename this var to something like:

```suggestion
  const THRESHOLD_OPACITY = 0.3;
```

It took me a while to see what this value was for exactly.